apps/emqx_utils/include/emqx_message.hrl

```diff
@@ -33,7 +33,7 @@
     %% Topic that the message is published to
     topic :: emqx_types:topic(),
     %% Message Payload
-    payload :: emqx_types:payload(),
+    payload,
```

Comment:
Is it possible to avoid this change? I'd prefer to relax the type in `emqx_ds` than to risk introducing a missed typing issue anywhere in emqx code.

Proper solution could look like this: remove all types from the record, and define `emqx_types:message()` as `#message{id :: .., payload :: .., ...}`.

Then define another type smth. like `message_match_pattern` that allows '_'.

Reply:
1) At least it would be nice to leave this type commented out with an explanation. 
2) But maybe it's worth to add another record just for precondition matching, say, `#message_matcher{}` with the same fields.